optimade/filtertransformers/mongo.py

```diff
@@ -389,6 +393,43 @@ def replace_with_relationship(subdict, prop, expr):
             filter_, check_for_entry_type, replace_with_relationship
         )
 
+    def _apply_unknown_or_null_filter(self, filter_: dict) -> dict:
+        """ This method loops through the query and replaces the check for
+        KNOWN with a check for existence and a check for not null, and the
+        inverse for UNKNOWN.
+
+        """
+
+        def check_for_known_filter(prop, expr):
+            """ Find cases where the query dict looks like
+            `{"field": {"#known": T/F}}` or
+            `{"field": "$not": {"#known": T/F}}`, which is a magic word
+            for KNOWN/UNKNOWN filters in this transformer.
+
+            """
+            return isinstance(expr, dict) and (
+                "#known" in expr or "#known" in expr.get("$not", {})
+            )
+
+        def replace_known_filter_with_or(subdict, prop, expr):
+            nor = set(expr.keys()) == {"$not"}
```

Comment:
Maybe change all instances of `nor` to `not_` or something similar instead? Or did I misinterpret the meaning?

Reply:
Ah, well spotted. Initially I was going directly to `$nor` rather than `$not`. Will change that.